src/main/java/org/killbill/billing/plugin/vertex/dao/VertexResponseDataExtractor.java

```diff
@@ -0,0 +1,328 @@
+
+        return additionalData;
+    }
+
+    private List<AddressInfo> buildAddresses() {
+        final CustomerType customer = vertexResponseData.getCustomer();
+        if (customer == null) {
+            return ImmutableList.of();
+        }
+
+        final List<AddressInfo> addresses = new LinkedList<>();
+
+        if (customer.getDestination() != null) {
+            addresses.add(new AddressInfo(customer.getDestination()));
+        }
+
+        if (customer.getTaxRegistrations() != null && customer.getTaxRegistrations().size() > 0) {
+            addresses.addAll(customer.getTaxRegistrations()
+                                     .stream()
+                                     .map(TaxRegistrationType::getPhysicalLocations)
+                                     .filter(Objects::nonNull)
+                                     .flatMap(Collection::stream)
+                                     .map(AddressInfo::new)
+                                     .collect(Collectors.toList()));
+        }
+
+        return addresses;
+    }
+
+    private BigDecimal computeTotalTaxCalculated() {
+        final List<OwnerResponseLineItemType> lineItems = vertexResponseData.getLineItems();
+        return lineItems != null
+               ? lineItems.stream()
+                          .filter(Objects::nonNull)
+                          .filter(lineItem -> lineItem.getTaxes() != null)
+                          .flatMap(lineItem -> lineItem.getTaxes().stream())
+                          .map(TaxesType::getCalculatedTax)
+                          .filter(Objects::nonNull)
+                          .map(BigDecimal::valueOf)
+                          .reduce(BigDecimal.ZERO, BigDecimal::add)
+               : BigDecimal.ZERO;
+    }
+
+    private BigDecimal computeTotalTaxExempt() {
+        final List<OwnerResponseLineItemType> lineItems = vertexResponseData.getLineItems();
+        return lineItems != null
+               ? lineItems.stream()
+                          .filter(Objects::nonNull)
+                          .filter(lineItem -> lineItem.getTaxes() != null)
+                          .flatMap(lineItem -> lineItem.getTaxes().stream())
+                          .map(TaxesType::getExempt)
+                          .filter(Objects::nonNull)
+                          .map(BigDecimal::valueOf)
+                          .reduce(BigDecimal.ZERO, BigDecimal::add)
+               : BigDecimal.ZERO;
+    }
+
+    private BigDecimal computeTotalTaxable() {
+        final List<OwnerResponseLineItemType> lineItems = vertexResponseData.getLineItems();
+        return lineItems != null
+               ? lineItems.stream()
+                          .filter(Objects::nonNull)
+                          .filter(lineItem -> lineItem.getTaxes() != null && lineItem.getTaxes().size() > 0)
+                          .map(lineItem -> lineItem.getTaxes().get(0))      //taxable field is the same for all taxes in the lineItem, so get the first one
```

Comment:
Not sure with this point as there are different taxable amounts, for example for billing-it for Colorado below are the values generated by Vertex:
![image](https://github.com/killbill/killbill-vertex-plugin/assets/93208596/b72108b5-b61a-4c7c-9029-13753ace060d)

Reply:
I think we just need to get the first non-null/non-zero value.

Reply:
Updated in latest commit and updated tests. Using nontaxable values for exempt.